test/toolTests.ts

```diff
@@ -69,6 +69,56 @@ describe('Tool Tests', function () {
         });
     });
 
+    it('downloads to an aboslute path', function () {
+        this.timeout(5000);
+
+        return new Promise<void>(async(resolve, reject)=> {
+            try {
+                let tempDownloadFolder: string = 'temp_' + Math.floor(Math.random() * 2000000000);
+                let aboslutePath: string = path.join(tempPath, tempDownloadFolder);
+                let downPath: string = await toolLib.downloadTool("http://httpbin.org/bytes/100", aboslutePath);
+                toolLib.debug('downloaded path: ' + downPath);
+                
+                assert(tl.exist(downPath), 'downloaded file exists');
+                assert(aboslutePath == downPath);
+
+                resolve();
+            }
+            catch(err) {
+                reject(err);
+            }
+        });
+    });
+
+    it('has status code in exception dictionary for non 200 response', function () {
+        this.timeout(2000);
+
+        return new Promise<void>(async(resolve, reject)=> {
+            try {
+                let nonExistentUrl: string = "http://httpbin.org/nonexistent";
+                let downPath: string = await toolLib.downloadTool(nonExistentUrl);
+
+                reject('a file was downloaded but it shouldnt have been');
+            } 
+            catch (err){
+                let statusCodeExists: boolean = false;
+
+                for (let errorKey of Object.keys(err))
```

Comment:
Fixed! Was looking for something like ContainsKey to do it safely, didn't realize you could do it this way in ts. Thanks!